webextension/background/translation.js

```diff
@@ -0,0 +1,46 @@
+var authToken;
+var tokenttl = 1000*60*10;
+var tokenTimeout = 0;
+
+sdk.runtime.onMessage.addListener((data, sender, sendReply) => {
+    if (data.action !== "translate") {
+        return;
+    }
+
+    chrome.storage.local.get({ "translation-secret-key": "" }, (prefs) => {
+        if (!prefs["translation-secret-key"] || prefs["translation-secret-key"] == "") {
+            sendReply({ error: 'Secret key not set' });
+            return;
+        }
+
+        if(tokenTimeout < Date.now()) {
+            var xhr = new XMLHttpRequest();
+            xhr.open("POST", "https://api.cognitive.microsoft.com/sts/v1.0/issueToken", true);
+            xhr.setRequestHeader('Ocp-Apim-Subscription-Key', prefs["translation-secret-key"]);
+            xhr.onreadystatechange = function() {
+                if (xhr.readyState == 4) {
+                    authToken = xhr.responseText;
+                    tokenTimeout = Date.now() + tokenttl;
+                    translate(data.text, sendReply);
+                }
+            };
+            xhr.send();
+        }
+        else {
```

Comment:
There should be an .eslintrc in the base. Can you run eslint to make sure style is ok?